From 9383091582ce635c6ea23e27228267a40de7042d Mon Sep 17 00:00:00 2001
From: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Date: Wed, 10 Oct 2012 11:35:19 +0200
Subject: memory init to avoid valgrind warnings

BZ: 61459

To avoid a valgrind warning, it is necessary to initialize the
allocated memory in Parcel.cpp This initialization should
avoid also an unpredictable behaviour because 0 is more
appropriate value that a random value from heap.
The issue has been found by valgrind when testing of mediaserver.

Category: Device enablement
Domain: AOSP-Bionic
Origin: Internal
Upstream-candidate: Yes

Change-Id: I54ba6b23cbf6738ab35647a378e81dc2357989b1
Orig-Change-Id: Ie2b4cc3fbdbc217fa4408c7bc3e9cb47b5d0d40e
Signed-off-by: Andrei V. FOMITCHEV <andreix.fomitchev@intel.com>
Signed-off-by: Jin Wei <wei.a.jin@intel.com>
---
 libs/binder/Parcel.cpp |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 38e019c..ca177f3 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -1481,7 +1481,8 @@ status_t Parcel::continueWrite(size_t desired)
         size_t* objects = NULL;
         
         if (objectsSize) {
-            objects = (size_t*)malloc(objectsSize*sizeof(size_t));
+            //valgrind warning fix: it is necessary to initialize the allocated memory
+            objects = (size_t*)calloc(objectsSize, sizeof(size_t));
             if (!objects) {
                 free(data);
 
@@ -1560,7 +1561,8 @@ status_t Parcel::continueWrite(size_t desired)
         
     } else {
         // This is the first data.  Easy!
-        uint8_t* data = (uint8_t*)malloc(desired);
+        //valgrind warning fix: it is necessary to initialize the allocated memory
+        uint8_t* data = (uint8_t*)calloc(desired, sizeof(uint8_t));
         if (!data) {
             mError = NO_MEMORY;
             return NO_MEMORY;
-- 
1.7.4.1

